src/repr/scalar/datetime.rs

```diff
@@ -890,159 +896,146 @@ fn fill_pdt_from_tokens(
+        match (atok, etok) {
+            // The following forms of punctuation signal the end of a field and can
+            // trigger a write.
+            (Dash, Dash) | (Colon, Colon) => {
+                pdt.write_field_iff_none(current_field, unit_buf)?;
+                unit_buf = None;
+                current_field = current_field.next_smallest();
+            }
+            (Delim, Delim) => {
+                pdt.write_field_iff_none(current_field, unit_buf)?;
+                unit_buf = None;
+                current_field = current_field.next_smallest();
+
+                // Spaces require special processing to allow users to enter an
+                // arbitrary number of delimiters wherever they're allowed. Note
+                // that this does not include colons.
+                actual.pop_front();
+                expected.pop_front();
+                i += 1;
 
-                    // Spaces require special processing to allow users to enter an arbitrary
-                    // number of spaces wherever spaces are allowed.
+                while let Some(Delim) = actual.front() {
                     actual.pop_front();
-                    expected.pop_front();
-                    i += 1;
-
-                    while let Some(Space) = actual.front() {
-                        actual.pop_front();
-                    }
-
-                    continue;
                 }
-                (TimeUnit(f), TimeUnit(_)) if unit_buf.is_some() => {
-                    if *f != current_field {
-                        failure::bail!(
+
+                continue;
+            }
+            (TimeUnit(f), TimeUnit(_)) if unit_buf.is_some() => {
+                if *f != current_field {
+                    failure::bail!(
                             "Invalid syntax at offset {}: provided TimeUnit({}) but expected TimeUnit({})",
                             i,
                             f,
                             current_field
                         );
-                    }
                 }
-                (TimeUnit(f), TimeUnit(_)) if unit_buf.is_none() => {
-                    failure::bail!(
-                        "Invalid syntax: {} must be preceeded by a number, e.g. \'1{}\'",
-                        f,
-                        f
-                    );
+            }
+            (TimeUnit(f), TimeUnit(_)) if unit_buf.is_none() => {
+                failure::bail!(
+                    "Invalid syntax: {} must be preceeded by a number, e.g. \'1{}\'",
+                    f,
+                    f
+                );
+            }
+            // Dots do not denote terminating a field, so should not trigger a write.
+            (Dot, Dot) => {}
+            (DateTimeDelimiter, DateTimeDelimiter) => {}
+            // avoid removing actual token if encountering non-matching DateTimeDelimiter
+            (_, DateTimeDelimiter) => {
```

Comment:
I actually removed `DateTimeDelimiter` from all of the `expected` tokens, so this can get nixed entirely. This was leftover special casting that allowed us to break out of processing token queues when encountering an unmatched `DateTimeDelimiter`.